core/templates/dev/head/editor/StateFallbacks.js

```diff
@@ -0,0 +1,195 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controllers, services and filters for fallbacks.
+ *
+ * @author sll@google.com (Sean Lip)
+ */
+
+oppia.controller('StateFallbacks', [
+    '$scope', '$rootScope', '$modal', '$filter', 'editorContextService',
+    'warningsData', 'INTERACTION_SPECS', 'stateFallbacksService',
+    'explorationStatesService', 'stateInteractionIdService',
+    function(
+      $scope, $rootScope, $modal, $filter, editorContextService,
+      warningsData, INTERACTION_SPECS, stateFallbacksService,
+      explorationStatesService, stateInteractionIdService) {
+  $scope.editorContextService = editorContextService;
+  $scope.stateFallbacksService = stateFallbacksService;
+  $scope.activeFallbackIndex = null;
+
+  $scope.$on('stateEditorInitialized', function(evt, stateData) {
+    stateFallbacksService.init(
+      $scope.stateName, stateData.interaction.fallbacks,
+      stateData.interaction, 'fallbacks');
+
+    $scope.activeFallbackIndex = null;
+  });
+
+  $scope.getFallbackSummary = function(fallback) {
+    var numSubmits = fallback.trigger.customization_args.num_submits.value;
+    var fallbackDescription = (
+      '[' + numSubmits +
+      ' unsuccessful attempt' + (numSubmits !== 1 ? 's' : '') + '] ');
+    var feedbackAsPlainText = (
+      fallback.outcome.feedback.length ?
+      $filter('convertToPlainText')(fallback.outcome.feedback[0]) :
+      '');
+    return fallbackDescription + feedbackAsPlainText;
+  };
+
+  $scope.changeActiveFallbackIndex = function(newIndex) {
+    // If the current fallback is being clicked on again, close it.
+    if (newIndex === $scope.activeFallbackIndex) {
+      $scope.activeFallbackIndex = null;
+    } else {
+      $scope.activeFallbackIndex = newIndex;
+    }
+  };
+
+  // This returns false if the current interaction ID is null.
+  $scope.isCurrentInteractionLinear = function() {
+    var interactionId = stateInteractionIdService.savedMemento;
+    return interactionId && INTERACTION_SPECS[interactionId].is_linear;
+  };
+
+  $scope.openAddFallbackModal = function() {
+    warningsData.clear();
+    $rootScope.$broadcast('externalSave');
+
+    $modal.open({
+      templateUrl: 'modals/addFallback',
+      backdrop: true,
+      controller: [
+          '$scope', '$modalInstance', 'editorContextService',
+          function($scope, $modalInstance, editorContextService) {
+
+        $scope.INT_FORM_SCHEMA = {
```

Comment:
This seems like the kind of thing to handle using the existing rule architecture rather than hard-coding it.